### ISSUE TITLE
Compiling latest Ifcopenshell v0.7.0 on windows -> IfcPythonPYTHON_wrap.cxx.obj : error LNK2001

### ISSUE THREAD

**Issue by Krande** [IfcOpenShell/IfcOpenShell#1949]
Hi,

I am experimenting (and learning) how to compile Ifcopenshell from the latest master-branch using cmake on github actions using conda-build for different os'es. 

And when trying to compile for windows, it seems like I consistently get the same error message:

When using NMake
```
[ 98%] Linking CXX shared module _ifcopenshell_wrapper.pyd
..
IfcPythonPYTHON_wrap.cxx.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
   Creating library _ifcopenshell_wrapper.lib and object _ifcopenshell_wrapper.exp
IfcPythonPYTHON_wrap.cxx.obj : error LNK2001: unresolved external symbol "public: class H5::IdComponent & __cdecl H5::IdComponent::operator=(class H5::IdComponent const &)" (??4IdComponent@H5@@QEAAAEAV01@AEBV01@@Z)
```

When using Ninja
```
[137/138] Linking CXX shared module ifcwrap\_ifcopenshell_wrapper.pyd
FAILED: ifcwrap/_ifcopenshell_wrapper.pyd 
...
IfcPythonPYTHON_wrap.cxx.obj : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
   Creating library ifcwrap\_ifcopenshell_wrapper.lib and object ifcwrap\_ifcopenshell_wrapper.exp
IfcPythonPYTHON_wrap.cxx.obj : error LNK2001: unresolved external symbol "public: class H5::IdComponent & __cdecl H5::IdComponent::operator=(class H5::IdComponent const &)" (??4IdComponent@H5@@QEAAAEAV01@AEBV01@@Z)
```

I have tried a lot of variations in the process, but here is an example of my cmake build configs:

```
mkdir build && cd build

cmake -G "Ninja" ^
 -D CMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
 -D CMAKE_BUILD_TYPE=Release ^
 -D CMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
 -D CMAKE_SYSTEM_PREFIX_PATH="%LIBRARY_PREFIX%" ^
 -D Python3_FIND_STRATEGY=LOCATION ^
 -D Python3_FIND_REGISTRY=NEVER ^
 -D OCC_INCLUDE_DIR="%LIBRARY_PREFIX%\include\opencascade" ^
 -D OCC_LIBRARY_DIR="%LIBRARY_PREFIX%\lib" ^
 -D CGAL_INCLUDE_DIR="%LIBRARY_PREFIX%\include" ^
 -D GMP_LIBRARY_DIR:FILEPATH="%LIBRARY_PREFIX%\lib" ^
 -D MPFR_LIBRARY_DIR:FILEPATH="%LIBRARY_PREFIX%\lib" ^
 -D HDF5_INCLUDE_DIR:FILEPATH="%LIBRARY_PREFIX%\include" ^
 -D HDF5_LIBRARY_DIR:FILEPATH="%LIBRARY_PREFIX%\lib" ^
 -D COLLADA_SUPPORT=OFF ^
 -D BUILD_EXAMPLES=OFF ^
 -D BUILD_GEOMSERVER=ON ^
 -D BUILD_CONVERT=OFF ^
 -D BUILD_IFCPYTHON=ON ^
 -D BUILD_IFCGEOM=ON ^
 -D IFCXML_SUPPORT=ON ^
 -D MSVC_EXTRA_WARNINGS=OFF ^
 %SRC_DIR%/cmake
if errorlevel 1 exit 1

:: Build and install
cmake --build . -- install

if errorlevel 1 exit 1
```

and my meta.yaml file looks like this:

```yaml
{% set name = "ifcopenshell" %}
{% set version = "0.7.0" %}
{% set occt_version = "7.5.3" %}

package:
  name: {{ name }}
  version: {{ version }}

source:
  git_url: https://github.com/IfcOpenShell/IfcOpenShell

build:
  binary_relocation: false [osx]

requirements:
  build:
    - {{ compiler('cxx') }}
    - ninja ==1.10.2
    - cmake
    - swig ==4.0.2

  host:
    - python {{ python }}
    - occt =={{ occt_version }}
    - cgal-cpp ==5.3

  run:
    - python
    - occt =={{ occt_version }}
    - lark-parser
    - deepdiff
    - cgal-cpp ==5.3

test:
  imports:
    - ifcopenshell

about:
  home: http://ifcopenshell.org
  license: LGPL-3.0-or-later
  license_file: COPYING
  summary: 'IfcOpenShell is a library to support the IFC file format'
  description: |
    IfcOpenShell is an open source (LGPL) software library for 
    working with the Industry Foundation Classes (IFC) file format.
  doc_url: http://ifcopenshell.org/
  dev_url: https://github.com/IfcOpenShell/IfcOpenShell

```

An example of this error using github actions is here: https://github.com/Krande/condapackaging/runs/4552522417?check_suite_focus=true#step:5:25307

While I continue to debug this, does anybody have an idea what could be generating this error and\or point me in the right direction for solving this?

Best Regards
Kristoffer (c++ n00b)

**Comment by aothms** [comment]
First of all, also see #1327 which is unfortunately still open

I think it's a bit in the ifcopenshell cmake (that may have been reported already somewhere else)

The issue seems to be that without ifcconvert, the serializers are not built, fine in 0.6, but in 0.7 the serializers are also required by python.  So these bits should be `BUILD_CONVERT OR BUILD_IFCPYTHON` 

* https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/cmake/CMakeLists.txt#L729-L746
* https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/cmake/CMakeLists.txt#L654-L659

It would be super awesome if you could test that and submit a separate PR for it :)

On an unrelated note, it seems to find the wrong Python includes

~~~
-- Found PythonLibs: C:/hostedtoolcache/windows/Python/3.10.0/x64/libs/python310.lib (found version "3.10.0") 
~~~

But the correct interpreter. Tbh these things can be so confusing in cmake

~~~
-- Found PythonInterp: C:/Miniconda/envs/conda-/conda-bld/ifcopenshell_1639684503627/_h_env/python.exe (found version "3.9.7") 
~~~

**Comment by Krande** [comment]
Thanks for the reply, @aothms! 

Okay, I will try with building the serializers see where that takes me :) 

I tried switching `BUILD_CONVERT=ON`, but it gave the following error
```
ninja: error: 'C:/Conda39/envs/conda-msvc/conda-bld/ifcopenshell_1639902973297/_h_env/Library/lib/libzlib.lib', needed by 'IfcConvert.exe', missing and no known rule to make it
```
I have tried adding `zlib` as a dependency in my conda build env, but it seems there is no `libzlib.lib` in my library. Is it a different library on conda perhaps?

Regarding a PR('s), would you like me to create a Github Actions CI for building ifcopenshell v0.7.0 daily snapshots on conda? 

And if necessary, I can try to make a PR for any necessary changes to the ifcopenshell cmake?

I will keep you updated on progress!

**Comment by aothms** [comment]
> but it seems there is no libzlib.lib in my library. Is it a different library on conda perhaps?

Yes, I'm afraid so. It's probably called just zlib.lib (or maybe even z.lib??). I think I have only seen the HDF5 install script (which is what the cmake script is currently targetted at) generate the name libzlib.lib. So I guess you'd have to make that conditionally in cmakelists.txt, just see which of these you can find.

All PR's welcome definitely! But ideally coordinate with @tpaviot #1327 so that we get the best of both worlds.

**Comment by Krande** [comment]
Hey @aothms! 

I recently picked up where I left off before christmas, and I think I managed to get ifcopenshell compiling on windows and build a conda package using github actions. (see the ci results for python 3.10 [here](https://github.com/Krande/IfcOpenShell/runs/5105022954?check_suite_focus=true)). Regarding coordinating with @tpaviot I have tried to use [his branch](https://github.com/tpaviot/IfcOpenShell/tree/azure-ci/conda) as a basis for my conda build setup. But I have probably made a mess of it during my experimentation (and lack of competence with cmake) :) (so there ought to be room for much improvement in my fork).

In the end I had to do some minor tweaks to zlib names in the CMakeLists.txt ([here](https://github.com/Krande/IfcOpenShell/blob/b6007c4117fbcdf046bf36f6b1d359d4958afdd9/cmake/CMakeLists.txt#L448-L454)) for the HDF5_LIBRARIES. And of course I had to rewrite the [bld.bat](https://github.com/Krande/IfcOpenShell/blob/v0.7.0-ada-dev/conda/bld.bat) and the [meta.yaml](https://github.com/Krande/IfcOpenShell/blob/v0.7.0-ada-dev/conda/meta.yaml)

I'll be honest and say that I do not know the consequences of changing the HDF5_LIBRARIES variable, so I will hold off on creating a PR and instead try to fix the conda compilations for linux/osx and do some more tests of the compiled packages.

Once that is done I would like to delve a bit deeper into the c++ code. So one idea I had was to investigate if I can add support for occt==7.6.0!

A side note regarding the overall build strategy. I was thinking whether or not it would be possible to utilize the compilation strategy used in [ci.yml](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/.github/workflows/ci.yml)? There we use a  cache for the c++ compilation steps and it should be doable to use the results to build a conda package? I would suspect that ought to be significantly faster. For me, the challenge would be how to create a conda package based on it (as I only have experience using conda-build). 


Best Regards
Kristoffer

**Comment by Krande** [comment]
Hi again, @aothms! Here's a small update: I have now also successfully compiled IfcOpenShell for linux using github actions and conda-build (see the resulting files [here](https://anaconda.org/krande/ifcopenshell/files). Both win/linux compiled versions pass all tests located in [src/ifcopenshell-python/test](https://github.com/Krande/IfcOpenShell/tree/v0.7.0-ada-dev/src/ifcopenshell-python/test).

So now only macOS remains (see my latest github workflow [here](https://github.com/Krande/IfcOpenShell/runs/5161352591?check_suite_focus=true) using [this github actions workflow](https://github.com/Krande/IfcOpenShell/blob/v0.7.0-ada-dev/.github/workflows/ci-daily-build.yml). 

Currently I am getting an error "C compiler ... is not able to compile a simple test program." from [here](https://github.com/Krande/IfcOpenShell/runs/5161352483?check_suite_focus=true#step:5:212).

I don't have much experience with compiling on macOS, but I'll just keep trying :) If you have any tips for compiling on macOS I would appreciate any hints :) 

Best Regards
Kristoffer

**Comment by Krande** [comment]
Hey, 

I believe I figured the previous issue. 

So now everything seems to compile just fine, but it takes forever (>6 hours). So the github action workflows stops before it finishes:(

See here: https://github.com/Krande/IfcOpenShell/actions/runs/1834355503

To solve the previous issue I just had to set the correct osx SDK version

The commit that solved it: https://github.com/Krande/IfcOpenShell/commit/c381679960774c4d73e57b897d2e4b543030884d


The issue I am facing now seems to have something to do with the linker by looking at the output here:

https://github.com/Krande/IfcOpenShell/runs/5168612897?check_suite_focus=true#step:5:1766

(writing from my phone, so excuse my brevity)

Best Regards
Kristoffer

**Comment by Krande** [comment]
Hi again, 

I finally have some time to continue where I left off.

Since last time I have tried a few different things; 

* Vary different XCode versions (with no success unfortunately)
* Tried older OSX v10.15 catalina (but I ran into some new issues, so I paused exploring this further for now)


Some things I am testing right now is to move from Ninja to Make and use a more verbose logging to see if that generates some more usable logging feedback.

From what I can tell from studying the available logs is that the compilation step is fine. It is however in the packaging step it fails. 

More precisely at the line:

`['usr/lib/libbsm.0.tbd', 'System/Library/PrivateFrameworks/IAP.framework/IAP.tbd']`

which is the last printed line before the whole slew of error messages `is not an ELF` that is repeated until the github actions runner runs out of its allotted time of 6 hours.

Here is the log from the packaging step

```
Packaging ifcopenshell
INFO:conda_build.build:Packaging ifcopenshell-0.7.0-py310h362fd55_0
Packaging ifcopenshell-0.7.0-py310h362fd55_0
Partially parsed some '.tbd' files in sysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk, pretending .tbds are their install-names
Adding support to 'conda-build' for parsing these in 'liefldd.py' would be easy and useful:
['usr/lib/libbsm.0.tbd', 'System/Library/PrivateFrameworks/IAP.framework/IAP.tbd'] ...
WARNING:conda_build.post:Partially parsed some '.tbd' files in sysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk, pretending .tbds are their install-names
Adding support to 'conda-build' for parsing these in 'liefldd.py' would be easy and useful:
['usr/lib/libbsm.0.tbd', 'System/Library/PrivateFrameworks/IAP.framework/IAP.tbd'] ...
 is not an ELF
```

inbetween a lot of the `is not an ELF` log messages I find random occurrences of 

``` 
 is not an ELF
 is not an ELF
Huge number of relocations (#149945). On the first #65535 will be parsed
 is not an ELF
```

followed by a few instances of 

```
 is not an ELF
 is not an ELF
Command 'LINKER_OPTIMIZATION_HINT' not parsed!
 is not an ELF
Command 'LINKER_OPTIMIZATION_HINT' not parsed!
 is not an ELF
```

and further down I find

```
 is not an ELF
Relocation #1 of __text seems corrupted
Relocation #2 of __text seems corrupted
Relocation #4 of __text seems corrupted
Relocation #5 of __text seems corrupted
 is not an ELF
 is not an ELF
```

the latter is intermittenly repeated until the end.

The github actions log source: 
https://github.com/Krande/condapackaging/runs/5818383646?check_suite_focus=true#step:5:1932


I'll try to devote some time on this issue as I would like make a PR for daily conda builds of ifcopenshell.

Alternatively I can create the PR now and we can try to solve the missing macOS compilation in the PR itself. 

Let me know what works best for you!

Best Regards
Kristoffer

**Comment by aothms** [comment]
1.

> From what I can tell from studying the available logs is that the compilation step is fine. It is however in the packaging step it fails.

2.

> Some things I am testing right now is to move from Ninja to Make and use a more verbose logging to see if that generates some more usable logging feedback.

If 1. is true (I also think it is) then 2. doesn't seem like it'd be the first thing to try. It really seems conda-build to completely barf on something so probably not related to ninja or make.

Did you try and compare with this:

> Maybe also cross reference with the config from @tpaviot https://github.com/tpaviot/IfcOpenShell/blob/azure-ci/conda/build.sh https://github.com/tpaviot/IfcOpenShell/blob/azure-ci/azure-pipelines.yml

Or maybe an older/newer version of conda-build?

Thanks for your continued effort on this. No preference on PR now or later, we can indeed just start without OSX. It's just so odd, that nobody else is running into this.

>           { name: macOS, distver: macos-latest, compiler: gcc },

Did you try `compiler:clang` btw?  That'd be the default compiler on recent osx.

**Comment by aothms** [comment]
Hm, I see the compiler found by cmake **is** Clang.

~~~
-- The C compiler identification is Clang 13.0.1
-- The CXX compiler identification is Clang 13.0.1
~~~

**Comment by Krande** [comment]
Thanks for the quick response!

Yes, I forgot to mention in my previous email, that yes I used the work by @tpaviot as a basis for my build.sh. However it is likely that I have messed something up in the process of experimenting by switching different flags on/off in an effort to debug :)

I do believe however that I have reverted back to the original files and ended up where I am now on a few occassions without any luck.

The "compiler: gcc" in my github actions ci file,  is no longer used. I tried testing different compilers a while back, but just forgot to remove it. I do believe I am using clang by looking at the compilation logging:

```
-- The C compiler identification is Clang 13.0.1
-- The CXX compiler identification is Clang 13.0.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: $BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: $BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
```

You might be correct regarding with changing to make from Ninja not really being the right medicine.

I noticed in the compilation for pythonocc-core @tpaviot did some tricks at the end for macOS:

```
# fix rpaths
if [ $(uname) == Darwin ]; then
    for lib in $(ls $SP_DIR/OCC/_*.so); do
      install_name_tool -rpath $PREFIX/lib @loader_path/../../../ $lib
    done
fi
```

which led me to this: https://github.com/conda/conda-build/issues/145

So in my latest attempt I will try to apply the same method on the ifcopenshell package:

```
cmake -G Ninja \
   -DCMAKE_BUILD_TYPE=Release \
   -DCMAKE_INSTALL_PREFIX=$PREFIX \
    ${CMAKE_PLATFORM_FLAGS[@]} \
   -DCMAKE_PREFIX_PATH=$PREFIX \
   -DCMAKE_SYSTEM_PREFIX_PATH=$PREFIX \
   -DPython3_FIND_STRATEGY=LOCATION \
   -DPython3_FIND_FRAMEWORK=NEVER \
   -DGMP_LIBRARY_DIR=$PREFIX/lib \
   -DMPFR_LIBRARY_DIR=$PREFIX/lib \
   -DOCC_INCLUDE_DIR=$PREFIX/include/opencascade \
   -DOCC_LIBRARY_DIR=$PREFIX/lib \
   -DHDF5_SUPPORT:BOOL=ON \
   -DHDF5_INCLUDE_DIR=$PREFIX/include \
   -DHDF5_LIBRARY_DIR=$PREFIX/lib \
   -DJSON_INCLUDE_DIR=$PREFIX/include \
   -DCGAL_INCLUDE_DIR=$PREFIX/include \
   -DCOLLADA_SUPPORT:BOOL=OFF \
   -DBUILD_EXAMPLES:BOOL=OFF \
   -DIFCXML_SUPPORT:BOOL=OFF \
   -DBUILD_CONVERT:BOOL=ON \
   -DBUILD_IFCPYTHON:BOOL=ON \
   -DBUILD_IFCGEOM:BOOL=ON \
   -DBUILD_GEOMSERVER:BOOL=OFF \
   -DIFCXML_SUPPORT:BOOL=OFF \
   -DBOOST_USE_STATIC_LIBS:BOOL=OFF \
   -DCMAKE_OSX_SYSROOT=/Applications/Xcode_13.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk \
   ./cmake

  # Build and install
  cmake --build . -- install

  # fix rpaths
  for lib in $(ls $SP_DIR/ifcopenshell/_*.so); do
    install_name_tool -rpath $PREFIX/lib @loader_path/../../../ $lib
  done
```

**Comment by Krande** [comment]
Okay, small update! 

I think I am closer to solving the issue with failing mac compilation using conda-build now. The problem with the conda post-build step seems to be related to [this issue](https://github.com/conda/conda-build/issues/3684) which causes the post build step to take a really long time consequently leaving the github actions runner to reach its maximum time limit.

So as you suspected this has nothing to do with the compilation itself. I will try to switch to a different conda-build version and/or test all the different suggested solution in that issue tracker.

I'll keep you posted. 

Best Regards
Kristoffer

**Comment by aothms** [comment]
Awesome, I didn't see it in my searches because I specifically looked for the text output, I thought the build was just broken, but apparently the issue is simply performance related, never would have guessed that with 6 hours!

I also see the cadquery developer reporting there, so it could be something specific to opencascade and some other large libraries perhaps.

Thanks!

**Comment by Krande** [comment]
Okay, I finally got it working!

https://github.com/Krande/condapackaging/runs/5855778279?check_suite_focus=true

and package here:
![image](https://user-images.githubusercontent.com/9642232/162051676-0242c653-5bca-4b99-909a-a51c5dd15172.png)

I will create a PR tomorrow so that we can get to work on daily builds uploaded to conda.

I see that there are 2 main repos for ifcopenshell on anaconda; conda-forge and ifcopenshell:
![image](https://user-images.githubusercontent.com/9642232/162052082-d8919b23-3a56-4e40-bf41-506d3fda0c2d.png)

Any preference on which we should use?

Personally I am not familiar with conda-forge so I would propose we revive the ifcopenshell repo and upload the daily builds there? Or I can just update the conda-forge repo (which would be fine, but it might take me some extra time).

Also, I would suggest we compile new versions of occt v7.5.3 so that it is compatible with python 3.10 (I've done that on my on anaconda repo to test that it works). Or we could jump straight to occt 7.6 (I noticed some commits referencing v7.6) which I believe is already compiled with dependencies compatible with python 3.10.

Let me know what you think,

Best Regards
Kristoffer

**Comment by aothms** [comment]
Woohoo great work.

> Personally I am not familiar with conda-forge so I would propose we revive the ifcopenshell repo and upload the daily builds there?

Agreed. It's a long time, since I published on anaconda. Let me know what you need. I'd prefer to just put you in charge of the ifcopenshell anaconda channel and enable you to manage secrets on the repo here if that's ok with you.

> Also, I would suggest we compile new versions of occt v7.5.3 so that it is compatible with python 3.10

I don't understand this bit. OCCT is not a python module, it's just the OCCT library and header files. It shouldn't depend on a python version. Are we talking about this one: https://anaconda.org/conda-forge/occt/files ? Do you think this is something that can be resolved upstream?

In general yes, that can be the downside of relying on conda-forge, old recipes are afaict not republished for newer python versions.

Ideally though we shouldn't recreate occt packages. We can switch to 7.6 if it saves us time and provides a more unified use of occt in the ecosystem.

**Comment by Krande** [comment]
Thanks!

Sure, I can handle the anaconda repo and secrets on github. Is the IfcOpenshell repo on Anaconda set up as an organization? I think that would be (so that we can have multiple administrators, and create access tokens linked to that organization and not directly to a user). 

Regarding the occt package; Yes, you are right, the occt package itself is python-version agnostic. But last time I checked it had some indirect dependencies that unfortunately were in conflict with python dependencies of ifcopenshell. But I agree with moving onto v7.6 and go from there! 

So if all goes well the installation using conda will be 

`conda install -c ifcopenshell -c conda-forge ifcopenshell`

If you want to create a separate channel for stable and development releases we could create a "dev" channel like so:

`conda install -c ifcopenshell/label/dev -c conda-forge ifcopenshell`

But that is something we can look into at a later stage.

Last but not least; do you have any preference on version names? 
I guess two options are that we simply overwrite the v0.7.0 version or add parts of the commit name to the version name. We can also discuss this in the PR I will create later today.

**Comment by aothms** [comment]
Good point, I think it's currently a user. Are you aware of a way of upgrading to an org (it's possible on github at least). I can't even find a way to delete.

> Last but not least; do you have any preference on version names?

We discussed this with @tpaviot on his issue as well:

> No, if the git number is appended to the version number, it won't be possible to install ifcopenshell==0.6.0, you will have to specify the git number as well, e.g. install ifcopenshell==0.6.0.bbccddeeff If the git commit number is appended as a suffix, then it will be possible to install any git commit, and roll back to a previous version in case of a regression. If ever the package version is only 0.6.0, then each new package overloads the older one. I would go with 0.6.0 only, no git number.

https://github.com/IfcOpenShell/IfcOpenShell/issues/1327#issuecomment-922428121

I agree and think we should just go with the plain branch name.

**Comment by aothms** [comment]
Sorry, that was embarrassing, somehow I had missed all of that. I've done as you directed and added you to the conda org. Let me know if you have any issues in setting up the secrets in the org on github. I haven't tried yet and I don't know the authorization requirements.

**Comment by Krande** [comment]
No worries. I see that the ifcopenshell organization is available through my anaconda dashboard. 

Next is that I need sufficient privileges to add secrets to the Ifcopenshell github repo. Or if you prefer I can create the access token and pm it to you and you add it to a secret on the github repo. Either way is fine for me :)

**Comment by Krande** [comment]
done. Sent it to your email. Let me know when you have added the secret.

**Comment by aothms** [comment]
Done

**Comment by Krande** [comment]
Great. PR created. We can continue this discussion in https://github.com/IfcOpenShell/IfcOpenShell/pull/2132 and feel free to close this issue
